docs/Guides/Create-Business-Users.md

```diff
@@ -93,14 +93,14 @@ If another user already exists within your platform with the same email address
 ```
 
 
-## Coming soon
+## Update business locations for an existing user
+You may add and remove a user's accessible business locations in the following way.
 
-### Link an existing user
-You may add one or more add additional business locations to an existing user in the following way.
+After this operation completes, the user will only have access to the specified locations.
 
 ```json http
 {
-  "method": "post",
+  "method": "patch",
```

Comment:
Is that previously documented POST with I guess just add semantics gone?
I might be confused: it seems like the caller is PUTing (replacing) the entire `businessLocations` resource (comprised of zero or many businessLocation objects), not PATCHing (partial updating)

Reply:
whoops yeah this was a miscopypasta 👍

Reply:
the previous example was WIP I believe, but I'm not actually sure what happens when POSTing instead of PATCHing

maybe if it's a POST the handler sees the set of (old+new) locations? cc @richard-rance

Reply:
there's definitely something different for POST vs PATCH. when I try running integration tests with POST instead of PATCH I'm seeing 404's for `/platform/users/<test userid>`

Reply:
it doesn't look like there are any endpoints that support both POST and PATCH (according to the platform.yaml anyway)

so maybe the thing to do here is remove the docs about POST for this?

Reply:
discord discussion: this seems like it should actually be a PUT. PATCH doesn't make too much sense it seems, since there's only 1 "field" being updated (the related ID's in the relationship)

but how can we tell what the request actually was from the API handler? by the time we get to the [`Update()` or `Create()` in the users resource](https://github.com/vendasta/api-gateway/blob/EA-277/users-update-relationships/internal/resource/users.go#L127), the lib has filled in values from a GET etc, right?

Reply:
Give [section 7.3 of the JSON:API spec](https://jsonapi.org/format/#crud-updating-relationships) a read.

PUT could be used for the complete replacement of the relationship data since it is the only field that exists at the subpath. The standard went with PATCH for consistency with update requests.

A POST to the subpath will append any ids on the request to what is currently there. It should ignore duplicates. 
A DELETE to the subpath will remove any ids on the request from what is currently there. Ids that do not exist should be ignored.

I did not write the OAS for the subpath since I was not sure if we would end up implementing it. You could just use the regular update path and only send the relationship field to achieve the value.

If you want to support POST and DELETE you will need to implement the [resource interface](https://github.com/vendasta/api-gateway/blob/f4b5ccec60c6728f2bec70fc7180ae1aaab06016/internal/resource/interfaces.go#L77-L80) and uncomment [this model code](https://github.com/vendasta/api-gateway/blob/865690fe35b71fbf4870f831ddee05da73f95565/internal/model/users.go#L220-L246) to implement the model interface.

---

You are right that by the time the `Update()` handler gets called the values have already been backfilled by a GET request. The way that update requests are handled is probably my least favorite part of the api2go library.

[This code](https://github.com/vendasta/api-gateway/blob/865690fe35b71fbf4870f831ddee05da73f95565/internal/storage/users/smb_role.go#L57-L67) will tell you what has changed.

Reply:
@richard-rance 
> If you want to support POST and DELETE you will need to implement the resource interface and uncomment this model code to implement the model interface.

the model side of that seems to make sense, I've implemented the `model.ResourceWithAdvancedToMany` iface but I don't understand the resource side of it--

how can that `resource.AdvancedUpdator` be implemented? the `resource.Users` struct isn't an instance of a user object, it's something like a service afaict. so given a method call to `user.AddToManyIDs("businessLocations", []string{"AG-1234")` how can that be implemented? (where `user` there refers to an instance of `resource.Users{}`)

Reply:
Looking at [the code](https://github.com/vendasta/api-gateway/blob/0f27191d21731764ede3132da7e84770b128f399/vendor/github.com/manyminds/api2go/api.go#L391) it is only the model that needs to implement the interface. We should remove the `resource.AdvancedUpdator` interface. I was wrong when adding it.